src/api.cc

```diff
@@ -58,17 +58,29 @@ const std::vector<boost::filesystem::path> AkliteClient::CONFIG_DIRS = {"/usr/li
 // It is applicable only to offline/local update.
 const static std::string LocalSrcDirKey{"local-src-dir"};
 
-TufTarget CheckInResult::GetLatest(std::string hwid) const {
+TufTarget CheckInResult::GetLatest(std::string hwid) const { return SelectTarget(-1, "", std::move(hwid)); }
+
+TufTarget CheckInResult::SelectTarget(int version, const std::string& target_name, std::string hwid) const {
   if (hwid.empty()) {
     hwid = primary_hwid_;
   }
 
-  for (auto it = targets_.crbegin(); it != targets_.crend(); ++it) {
-    if ((*it).Custom()["hardwareIds"][0] == hwid) {
-      return *it;
+  if (version == -1 && target_name.empty()) {
+    for (auto it = targets_.crbegin(); it != targets_.crend(); ++it) {
+      if ((*it).Custom()["hardwareIds"][0] == hwid) {
+        return *it;
+      }
+    }
+  } else {
+    for (const auto& t : targets_) {
+      if (t.Version() == version || t.Name() == target_name) {
```

Comment:
I think we need to match hwid too. There can be multiple targets of the same version but for different hardware ID.

Reply:
Actually, in the current implementation the `CheckInResult::targets_` are already filtered by hardware ID. So, we need either remove hwid checking for both cases, target name/version is specified and not or do checking for both cases.

Reply:
I'm now checking for hwid when version or name is specified too. So, it is now always checked.